sound/soc/sof/intel/hda.c

```diff
@@ -492,10 +492,15 @@ static int hda_init(struct snd_sof_dev *sdev)
 	/* HDA base */
 	sdev->bar[HDA_DSP_HDA_BAR] = bus->remap_addr;
 
-	/* init i915 and HDMI codecs */
-	ret = hda_codec_i915_init(sdev);
-	if (ret < 0)
-		dev_warn(sdev->dev, "init of i915 and HDMI codec failed\n");
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
+	/* do not init i915 codec if codec is masked */
+	if (HDA_IDISP_CODEC(hda_codec_mask)) {
+		/* init i915 and HDMI codecs */
+		ret = hda_codec_i915_init(sdev);
+		if (ret < 0)
+			dev_warn(sdev->dev, "init of i915 and HDMI codec failed\n");
+	}
+#endif
```

Comment:
the commit message should use a more professional tone

The SOF HDA driver goes through a 60s timeout if the i915 driver is missing or not fully functional in early silicon enablement. When we already have the information through the codec_mask that the display codec is not reported, we can avoid the i915 initialization altogether and skip the timeout.

Reply:
Thanks for the advice, I will apply this in update.